### PR TITLE
VIH-7802 StaffMember waitingroom

### PR DESCRIPTION
**Pull request by ZJianWang**
### JIRA link (if applicable) ###

VIH-7802

### Change description ###

Dual host waitingroom

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb.UnitTests/Mappings/ParticipantForUserResponseMapperTests.cs
Comment (FarronF) [inline VideoWeb/VideoWeb.UnitTests/Mappings/ParticipantForUserResponseMapperTests.cs:105]: Can this be figured out instead of needing to be changed in the test every time we update?

File: VideoWeb/VideoWeb/ClientApp/src/app/testing/mocks/data/conference-test-data.ts
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/testing/mocks/data/conference-test-data.ts:443]: I think we should in general move away from using this test data and build the data we need in the tests to keep them relevant. But it's fine here.

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.spec.ts
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.spec.ts:146]: This doesn't test it saves to DB just that we call the function on videoWebService.

Comment (ZJianWang) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.spec.ts:146]: I'll correct the name

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.ts
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.ts:32]: Why are these removed?

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.ts
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.ts:171]: This log says judge instead of staff member

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.ts
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.ts:178]: Change update to updating?

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.ts
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.ts:88]: Remove console log

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.ts
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.ts:33]: Any reason we have a list of staffMembers? My understanding was that it would be just 1

File: VideoWeb/VideoWeb/Extensions/ConfigureAuthSchemeExtensions.cs
Comment (FarronF) [inline VideoWeb/VideoWeb/Extensions/ConfigureAuthSchemeExtensions.cs:121]: QuickLink users should not be removed

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.ts
Comment (ZJianWang) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.ts:165]: I'm not sure we need it ot not, but the judge have representee as well

Comment (FarronF) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.ts:165]: Ah. I didn't see that. Seems odd for sure, I'll check again.

File: VideoWeb/VideoWeb/Extensions/ConfigureAuthSchemeExtensions.cs
Comment (ZJianWang) [reply VideoWeb/VideoWeb/Extensions/ConfigureAuthSchemeExtensions.cs:121]: I mis removed it when I fix the conflicts

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.ts
Comment (ZJianWang) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/wr-participant-list-shared.component.ts:33]: yeah I know, I can use another way to handle this

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.ts
Comment (ZJianWang) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.ts:32]: these are not used anywhere

File: VideoWeb/VideoWeb/Extensions/ConfigureAuthSchemeExtensions.cs
Comment (FarronF) [inline VideoWeb/VideoWeb/Extensions/ConfigureAuthSchemeExtensions.cs]: Remove line 122

File: VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/judge-hearing-list/judge-hearing-list.component.ts
Comment (robertbennett1998) [inline VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/judge-hearing-list/judge-hearing-list.component.ts:102]: Can we rename result to something more descriptive like "useJudgeWaitingRoom"

**Review by robertbennett1998 — DISMISSED** [review]
LGTM, minor comment.

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.spec.ts
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.spec.ts:168]: This is now excluding judge and staff member

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.spec.ts
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.spec.ts:111]: Why did this change from Dynamic to OnePlus7?

File: VideoWeb/VideoWeb/ClientApp/src/assets/i18n/en.json
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/assets/i18n/en.json:398]: Do we have Welsh or is that pending?

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.spec.ts
Comment (ZJianWang) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.spec.ts:111]: Beause I added a staff member to the participants list and the recomand layout depends on total numbers of participants

File: VideoWeb/VideoWeb/ClientApp/src/assets/i18n/en.json
Comment (ZJianWang) [reply VideoWeb/VideoWeb/ClientApp/src/assets/i18n/en.json:398]: yeah, I will ask Lorenzo and this will be added in another ticket

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.spec.ts
Comment (FarronF) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.spec.ts:111]: Is there test cases for different numbers of participants? This should really be separated from the ConferenceTestData if a change there will effect tests like this.

Comment (FarronF) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.spec.ts:111]: I see there's test cases for the number of participants. This more tests that the correct methods are called. I'd be inclined to change this by spying on the relevant methods and ensure they are called with the correct values. Since this is a change to existing code I'll approve and leave it up to you if you want to update.

**Review by FarronF — APPROVED** [review]

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.spec.ts
Comment (ZJianWang) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.spec.ts:111]: I think there is no test cases for different numbers of participants

**Comment by FarronF** [comment]
Looks good with the understanding Welsh translations will be added.

I have a comment regarding a unit test needing to be changed because of a change in the test data but due to our discussion about improving the code base I'm ok with the manual change here if you want to push this through.

**Review by robertbennett1998 — APPROVED** [review]
LGTM

**Pull request merged**
